src/components/connect.tsx

```diff
@@ -485,24 +463,9 @@ function connect<
 
   type WrappedComponentProps = TOwnProps & ConnectProps
 
-  const initMapStateToProps = match(
-    mapStateToProps,
-    // @ts-ignore
-    defaultMapStateToPropsFactories,
-    'mapStateToProps'
-  )!
-  const initMapDispatchToProps = match(
-    mapDispatchToProps,
-    // @ts-ignore
-    defaultMapDispatchToPropsFactories,
-    'mapDispatchToProps'
-  )!
-  const initMergeProps = match(
-    mergeProps,
-    // @ts-ignore
-    defaultMergePropsFactories,
-    'mergeProps'
-  )!
```

Comment:
The `match` function is not TypeScript friendly. I refactored those to just be functions (that I think are just as easy to understand as before).